### PR TITLE
add scalar formua

### PR DESCRIPTION
**Pull request by smilesun**
Hi, in this branch, I edited the FDboost to have the ability to add scalar vairiables for boosting.

@
@lbeggel @pfistfl , please review .

### CODE REVIEW

File: R/RLearner_FDAregr_FDboost.R [hunk 1]
```diff
@@ -62,6 +62,11 @@ trainLearner.fdaregr.FDboost = function(.learner, .task, .subset, .weights = NUL
     formula.terms[fdn] = sprintf("bsignal(%s, %s, knots = %i, df = %f, degree = %i, differences = %i, check.ident = %s)",
       fdn, gn, bsignal.knots, df, bsignal.degree, bsignal.differences, bsignal.check.ident)
   }
+  for (fsn in names(tdesc$fd.scalars)) {
```
Comment (lbeggel) [inline R/RLearner_FDAregr_FDboost.R]: tdesc$fd.scalars is never defined

File: R/RLearner_FDAregr_FDboost.R [hunk 2]
```diff
@@ -62,6 +62,11 @@ trainLearner.fdaregr.FDboost = function(.learner, .task, .subset, .weights = NUL
...
+    mat.list[[fsn]] = as.vector(as.matrix(d[, fsn, drop = FALSE]))
+    formula.terms[fsn] = sprintf("bbs(%s, knots = %i, df = %f, degree = %i, differences = %i)",
+      fsn, bsignal.knots, df, bsignal.degree, bsignal.differences)
+  }
```
Comment (lbeggel) [inline R/RLearner_FDAregr_FDboost.R]: can you comment on what the function does?

Comment (smilesun) [inline R/RLearner_FDAregr_FDboost.R]: ok

File: R/RLearner_FDAregr_FDboost.R (hunk 1 above)
Comment (smilesun) [inline R/RLearner_FDAregr_FDboost.R]: it is defined

File: R/RLearner_FDAregr_FDboost.R (hunk 2 above)
Comment (smilesun) [inline R/RLearner_FDAregr_FDboost.R]: done

**Pull request merged**